src/Cardano/Wallet/Primitive.hs

```diff
@@ -176,6 +230,25 @@ instance Dom UTxO where
     type DomElem UTxO = TxIn
     dom (UTxO utxo) = Map.keysSet utxo
 
+balance :: UTxO -> Integer
```

Comment:
NOTE: `balance` isn't giving back a `Coin` because this may overflow. Hence, we'll have to handle that properly whenever we make use of that.